### PR TITLE
Remove jQuery AJAX from repo collaborator mode dropdown

### PR DESCRIPTION
**Pull request by yardenshoham**
- Removed all jQuery AJAX calls and replaced with our fetch wrapper
- Tested the repo collaborator mode dropdown functionality and it works as before

# Demo using `fetch` instead of jQuery AJAX
action

### CODE REVIEW

File: web_src/js/features/repo-settings.js [hunk 1]
```diff
@@ -11,18 +12,19 @@ export function initRepoSettingsCollaboration() {
     const $dropdown = $(e);
     const $text = $dropdown.find('> .text');
     $dropdown.dropdown({
-      action(_text, value) {
+      async action(_text, value) {
         const lastValue = $dropdown.attr('data-last-value');
```
Comment (delvh) [inline web_src/js/features/repo-settings.js:16]:
```suggestion
```

File: web_src/js/features/repo-settings.js [hunk 2]
```diff
@@ -11,18 +12,19 @@ export function initRepoSettingsCollaboration() {
...
+        try {
+          $dropdown.attr('data-last-value', value);
+          $dropdown.dropdown('hide');
+          const data = new FormData();
+          data.append('uid', $dropdown.attr('data-uid'));
+          data.append('mode', value);
+          await POST($dropdown.attr('data-url'), {data});
+        } catch {
           $text.text('(error)'); // prevent from misleading users when error occurs
           $dropdown.attr('data-last-value', lastValue);
```
Comment (delvh) [inline web_src/js/features/repo-settings.js:26]:
```suggestion
          $dropdown.dropdown('hide');
          const data = new FormData();
          data.append('uid', $dropdown.attr('data-uid'));
          data.append('mode', value);
          await POST($dropdown.attr('data-url'), {data});
          $dropdown.attr('data-last-value', value);
        } catch {
          $text.text('(error)'); // prevent from misleading users when error occurs
```
These two suggestions should be identical to the previous behavior.

**Comment by yardenshoham** [comment]
Last commit seems to break it

**Comment by delvh** [comment]
What do you mean?

**Comment by yardenshoham** [comment]
action

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**